src/impl/tokenParser.js

```diff
@@ -447,14 +447,19 @@ export class TokenParser {
     }
   }
 
-  explainFromTokens(input) {
+  explainFromTokens(input, strictHours = false) {
     if (!this.isValid) {
       return { input, tokens: this.tokens, invalidReason: this.invalidReason };
     } else {
       const [rawMatches, matches] = match(input, this.regex, this.handlers),
         [result, zone, specificOffset] = matches
           ? dateTimeFromMatches(matches)
           : [null, null, undefined];
+      if (strictHours && hasOwnProperty(matches, "h") && (matches.h > 12 || matches.h < 1)) {
+        throw new ConflictingSpecificationError(
```

Comment:
It should produce an invalid `DateTime`, just like `DateTime.fromFormat('99:23', 'hh:mm')` does currently.
`ConflictingSpecificationError` means the programmer misused `fromFormat`.